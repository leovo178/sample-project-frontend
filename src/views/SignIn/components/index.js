export { default as SignInFormEditor } from './SignInFormEditor'
