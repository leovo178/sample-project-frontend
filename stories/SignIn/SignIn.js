import React, { useState } from 'react'
import { Box, Button, TextField, Typography } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import PropTypes from 'prop-types'
import LinkTo from '@storybook/addon-links/react'
import { action } from '@storybook/addon-actions'
import { text } from '@storybook/addon-knobs'

const useStyles = makeStyles(theme => ({
	root: {
		width: '100%',
		display: 'flex',
		alignItems: 'center',
	},
	container: {
		width: 381,
		margin: '20px auto',
		padding: theme.spacing(2),
		display: 'flex',
		flexDirection: 'column',
		alignItems: 'center',
	},
	title: {
		color: theme.palette.primary.main,
		fontWeight: 600,
	},
	input: {
		width: '100%',
		marginTop: '18px',
	},
	cardcontent: {
		padding: 0,
	},
	actions: {
		padding: 0,
		width: '100%',
		'&>button': {
			marginTop: theme.spacing(2),
		},
	},
	button: {
		fontWeight: 600,
		textTransform: 'unset',
		padding: '18px 0',
	},
	signuptext: {
		marginTop: theme.spacing(2),
		cursor: 'pointer',
	},
}))

const SignIn = () => {
	const [email, setEmail] = useState('')
	const [password, setPassword] = useState('')

	const classes = useStyles()

	const onSignIn = () => {
		handleSignIn({ email, password })
	}
	const mail = text('email', 'example@gmail.com')
	const pass = text('pass', '12345')

	const handleSignIn = ({ email, password }) => {
		action('Sign-in')({ email, password })

		if (!email) {
			alert('empty email')
			return
		}
		if (!password) {
			alert('empty password')
			return
		}

		if (mail === email && pass === password) alert('Logged in')
		else alert('Invalid email or password')
	}

	return (
		<Box className={classes.root}>
			<Box className={classes.container}>
				<Typography variant='h5' className={classes.title}>
					Sign in
				</Typography>
				<div className={classes.cardContent}>
					<TextField
						value={email}
						label='EMAIL'
						variant='outlined'
						type='email'
						autoComplete='true'
						className={classes.input}
						onChange={e => setEmail(e.target.value)}
					/>
					<TextField
						value={password}
						label='PASSWORD'
						variant='outlined'
						type='password'
						autoComplete='true'
						className={classes.input}
						onChange={e => setPassword(e.target.value)}
					/>
				</div>
				<div className={classes.actions}>
					<Button
						data-cy='signin-button'
						variant='contained'
						color='primary'
						size='large'
						fullWidth
						className={classes.button}
						onClick={onSignIn}
					>
						Sign in
					</Button>{' '}
					<Typography
						data-cy='signup-text'
						variant='body2'
						className={classes.signuptext}
						// onClick={linkTo('SignUp', 'signUp')}
					>
						{' '}
						<LinkTo
							kind='component-api-signup'
							story='sign-up'
							// story='SignUp' // it only use when have more than 2 story in abcxyz.stories.js
							// // target='_blank'
							// style={{ color: '#1474f3' }}
						>
							sign up
						</LinkTo>
						<LinkTo
							kind='component-api-signup'
							story='sign-up'
							// story='SignUp' // it only use when have more than 2 story in abcxyz.stories.js
							// // target='_blank'
							// style={{ color: '#1474f3' }}
						>
							Reservation
						</LinkTo>
					</Typography>
				</div>
			</Box>
		</Box>
	)
}

export default SignIn

SignIn.propTypes = {
	history: PropTypes.object,
}

SignIn.defaultProps = {}
