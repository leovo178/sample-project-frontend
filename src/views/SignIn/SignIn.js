import React from 'react'
import { withRouter } from 'react-router-dom'
import { useMutation } from '@apollo/react-hooks'

import { SignInFormEditor } from './components'

import { setToken } from '@src/shares/utils'
import { SIGN_IN } from './gql/mutation'
import { SET_UPLOADED_FILE } from '@views/File/gql/mutation'

const SignIn = props => {
	const { history } = props

	const [setUploadedFile] = useMutation(SET_UPLOADED_FILE, {
		onError: err => alert(err),
	})

	const [signIn] = useMutation(SIGN_IN, {
		onCompleted: ({ login: { token, user } }) =>
			handleOnSignInCompleted({ user, token }),
		onError: err => alert(err),
		fetchPolicy: 'no-cache',
	})

	const handleOnSubmit = (email, password) => {
		signIn({ variables: { user: { email, password } } })
	}

	const navigateToHomePage = () => history.push('/')

	const handleOnSignInCompleted = ({ user, token }) => {
		setUploadedFile({ variables: { file: user.image } })
		setToken(token)
		navigateToHomePage()
	}

	return <SignInFormEditor history={history} onSubmit={handleOnSubmit} />
}

export default withRouter(SignIn)
