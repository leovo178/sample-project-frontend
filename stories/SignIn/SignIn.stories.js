import React from 'react'
import SignIn from './SignIn'
import { ThemeProvider } from '@material-ui/core'
import muiTheme from '../../src/theme/muiTheme'

export default {
	title: 'Component Api|SignIn',
	includeStories: [],
}

export const signIn = () => {
	return (
		<ThemeProvider theme={muiTheme}>
			<SignIn />
		</ThemeProvider>
	)
}
