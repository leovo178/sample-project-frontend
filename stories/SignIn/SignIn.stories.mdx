import { Meta, Story, Preview,Props } from '@storybook/addon-docs/blocks';
import SignIn from './SignIn'
import * as stories from './SignIn.stories.js';


<Meta title="Component Api|SignIn" />

# SignIn
With `MDX` we can define a story for `Sign In` right in the middle of our
markdown documentation.




  <Story name="SignIn">
    {stories.signIn()}
  </Story>


```jsx 

<Route path='/sign-in' component={SignIn} />
```

# PropTypes

<Props of={SignIn}/>

