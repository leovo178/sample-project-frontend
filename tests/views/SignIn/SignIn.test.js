import React from 'react'

import { act } from '@testing-library/react'

import SignIn from '@views/SignIn'

import { findDOMNodeOfSignIn } from './helpers'

describe('<SignIn />', () => {
	let rendered
	beforeEach(async () => {
		await act(async () => {
			rendered = findDOMNodeOfSignIn(<SignIn />)
		})
	})

	it('should match snapshot', async () => {
		expect(rendered.container).toMatchSnapshot()
	})
})
