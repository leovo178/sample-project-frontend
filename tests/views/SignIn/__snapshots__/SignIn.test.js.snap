// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-10 makeStyles-root-1"
  >
    <div
      class="MuiBox-root MuiBox-root-11 makeStyles-container-2"
    >
      <h5
        class="MuiTypography-root makeStyles-title-3 MuiTypography-h5"
      >
        Sign in
      </h5>
      <div>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-input-4"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
          >
            EMAIL
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="true"
              class="MuiInputBase-input MuiOutlinedInput-input"
              placeholder="Email"
              type="email"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-100 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-102"
              >
                <span>
                  EMAIL
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-input-4"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
          >
            PASSWORD
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="true"
              class="MuiInputBase-input MuiOutlinedInput-input"
              placeholder="Password"
              type="password"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-100 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-102"
              >
                <span>
                  PASSWORD
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-actions-6"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-7 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
          data-testid="signin-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Sign in
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <p
          class="MuiTypography-root makeStyles-signuptext-8 MuiTypography-body2"
          data-testid="signup-text"
        >
          sign up
        </p>
      </div>
    </div>
  </div>
</div>
`;
